src/main/java/org/jabref/logic/importer/fetcher/DoiFetcher.java

```diff
@@ -47,18 +51,28 @@ public String getName() {
     @Override
     public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
         Optional<DOI> doi = DOI.parse(identifier);
+        String agency = "";
         try {
             if (doi.isPresent()) {
```

Comment:
There is `.orElseThrow(...)` in optionals. Rewriting is difficult here. And it was like that before you touched the code, so no importance here.

Reply:
About .orElseThrow I left the code as I found it. In this method I just added the medra check. 
In some other parts as well I preferred to leave the code style I found.